src/include/lru.h

```diff
@@ -189,6 +189,10 @@ class LRU {
     //generic_dout(10) << "lru: " << lru_get_size() << " items, " << top.size() << " top, " << bottom.size() << " bot, " << pintail.size() << " pintail" << dendl;
   }
 
+  bool is_on_lru(LRUObject *o) {
+    return o->lru_link.is_on_list();
+  }
+
```

Comment:
Typically it's best to introduce new functions like this along with their first caller, so that we can see how it's intended to be used. Maybe squash this patch together with the first one that uses is_on_lru()?

Reply:
> Typically it's best to introduce new functions like this along with their first caller, so that we can see how it's intended to be used.

For this, the caller is in the inode lock patch, and only when the inode lock is introduced will the caller is needed. This is just to make sure during the `client_lock`'s unlock/lock gap, we won't release or change the object twice.

> Maybe squash this patch together with the first one that uses is_on_lru()?

Sure.